src/common/repo/github-search.ts

```diff
@@ -0,0 +1,57 @@
+import type { Repo } from ".";
+
+const SearchDataObj = {
+  query: "",
+  count: 0,
+  result: null,
+}
+
+export type SearchData = typeof SearchDataObj;
+
+export class GitHubSearch {
+  private repo: Repo;
+  private searchTerms: string[] = [];
+  private cache: object = {};
+
+  constructor(repo: Repo){
+    this.repo = repo;
+  }
+
+  public addSearchTerm(term: string) {
+    this.searchTerms.push(term)
+  }
+
+  public addSearchTerms(terms: string[]) {
+    this.searchTerms = this.searchTerms.concat(terms);
+  }
+
+  public getResultsFor(terms: string[]) : SearchData[] {
+    var results: SearchData[] = [];
+
+    terms.forEach(term => {
```

Comment:
Maybe we can do something like 
```suggestion
    return terms.
        filter(term => this.cache[term]).
        map(term => { ...SearchDataObj, ...this.cache[term]})
```
and then removing result at the top 

Instead